### PR TITLE
Add gift card purchase, preview, & get

### PR DESCRIPTION
**Pull request by shulmang**
Adds support for gift card purchasing

### CODE REVIEW

File: lib/recurly/resource.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ class Resource
     #     e.message # => "Can't find Account with account_code = tortuga"
     #   end
     class NotFound < API::NotFound
-      def initialize message
+      def initialize(message)
```
Comment (bhelx) [inline lib/recurly/resource.rb]: This is great, but is there any way you could put these changes in their own commit?

Comment (shulmang) [inline lib/recurly/resource.rb]: Good call, @bhelx .  Should be all set.

Comment (bhelx) [inline lib/recurly/resource.rb]: Thanks a lot. It's a minor thing but helpful for keeping blames clean.

**Comment by bhelx** [comment]
Closing in favor of https://github.com/recurly/recurly-client-ruby/pull/257
